src/components/implementation/no_interface/cFE_booter/osfilesys.c

```diff
@@ -0,0 +1,308 @@
+#include "cFE_util.h"
+#include "osfilesys.h"
+
+struct FS fs[MAX_NUM_FS];
+struct fsobj files[MAX_NUM_FILES];
+
+/*
+ * Notes on this version:
+ * This is not the final implementation of a file system for the cFE on COS
+ * For now the file system is read only and uses static memory for fsobj and FS
+ * The next step is to write a dumb allocator/deallocator and use it for 
+ * new fs, fsobj, and file data
+ * I have been intentional to restrict memory specific details to this file
+ * osfilesys.h and osfiles.c 'should' not care about how FS, fsobj, and data
+ * are stored in memory.
+ */
+
+/******************************************************************************
+** Tar Level Methods
+******************************************************************************/
+
+/* 
+ * Loads the position in memory of linked
+ * Checks for badly linked or no linked tar file.
+ */
+uint32 tar_load()
+{
+    if(! &_binary_cFEfs_tar_size)                         return OS_FS_ERR_DRIVE_NOT_CREATED;
+    if(!_binary_cFEfs_tar_start)                          return OS_FS_ERR_DRIVE_NOT_CREATED;
+    if(&_binary_cFEfs_tar_end < &_binary_cFEfs_tar_start) return OS_FS_ERR_DRIVE_NOT_CREATED;
+    size    = (uint32) &_binary_cFEfs_tar_size;
+    start   = &_binary_cFEfs_tar_start;
+    end     = &_binary_cFEfs_tar_end;
+    return OS_FS_SUCCESS;
+}
+
+/*
+ * parses a loaded tar into fs[0]
+ */
+uint32 tar_parse()
+{
+    if(!start || !end)              return OS_FS_ERR_DRIVE_NOT_CREATED;
+    if(end-start-size)              return OS_FS_ERR_DRIVE_NOT_CREATED;
```

Comment:
so the reason I did end-star-size is because comparisons and arithmetic among pointers not within the same object (array or struct) are undefined behavior.  The reason I did it originally was because `end-start-size` is does not give a warning while `size != end-start` does.  It looks like both are undefined behavior that our compiler happens to implement in the way you would expect.  Now it seems like I should find another way to test for the same state, although none immediately come to mind